[go] Improve the incompatible SDK version error screen (#24310)
expo/expo: docs/pages/get-started/expo-go.mdx

---
title: Expo Go
description: Learn about Expo Go, the free, open-source sandbox app for learning and experimenting with React Native on Android and iOS.
---

import ImageSpotlight from '~/components/plugins/ImageSpotlight';
import { Collapsible } from '~/ui/components/Collapsible';
import { Terminal } from '~/ui/components/Snippet';
import { BoxLink } from '~/ui/components/BoxLink';
import { BookOpen02Icon } from '@expo/styleguide-icons';
import { Step } from '~/ui/components/Step';
import { Tabs, Tab } from '~/ui/components/Tabs';

> [Expo Go](https://expo.dev/client) is a free, [open-source](https://github.com/expo/expo/tree/main/home) sandbox for learning and experimenting with React Native on Android and iOS devices. You can install it directly from App Stores, and get up and running within minutes — no need to install a native toolchain and compile an app.

Reviewer: ```suggestion
[Expo Go](https://expo.dev/client) is a free, [open-source](https://github.com/expo/expo/tree/main/home) sandbox for learning and experimenting with React Native on Android and iOS devices. You can install it directly from App Stores, and get up and running within minutes &mdash; no need to install a native toolchain and compile an app.
```

To use a dash, let's use `&mdash;` otherwise, we can also split the sentence.

> Expo Go is not recommended for building production-quality apps — use [development builds](/develop/development-builds/introduction.mdx) instead. It's also not recommend for deploying/distributing apps, [learn more about deployment](/deploy/build-project/). That said, everything that you learn while working in Expo Go will be transferable to development builds, and you're likely to come back to Expo Go along your journey when you want to try new ideas in a quick new project sandbox. It's a great place to start!

Reviewer: ```suggestion
Expo Go is not recommended for building production-quality apps &mdash; use [development builds](/develop/development-builds/introduction) instead. It's also not recommended for [deploying/distributing apps](/deploy/build-project/). That said, everything you learn while working in Expo Go will be transferable to development builds, and you're likely to come back to Expo Go along your journey when you want to try new ideas in a quick new project sandbox. It's a great place to start!
```

- replaced dash with mdash;
- Fix typo
- I'd suggest we put the link on the context instead of adding the context by "learn more...". It's makes it easier for readers to scan through quickly without grabbing too much of their attention.

<Collapsible summary="What if I open a project with an unsupported SDK version?">

When running a project that was created for an unsupported SDK version in Expo Go, you'll see the following error:
```
"Project is incompatible with this version of Expo Go"
```
To fix this, upgrading your project to a [supported SDK version](/versions/latest/#each-expo-sdk-version-depends-on-a-react-native-version) is highly recommended. If you want to learn how to do it, see [Upgrade the project to a new SDK Version](#upgrade-the-project-to-a-new-sdk-version).

</Collapsible>

<Collapsible summary="How do I upgrade my project from an unsupported SDK version?">

Refer to the ["Upgrading Expo SDK" guide](/workflow/upgrading-expo-sdk-walkthrough.mdx) for instructions for upgrading to a specific SDK version.

</Collapsible>

<Collapsible summary="How can I install an older version of Expo Go to run my project?">

> If you can't upgrade your project to a newer SDK version, you can still run it using an older version of Expo Go — however, this method will not work for physical iOS devices because it is not possible to install apps directly (sideload). If you need to run your project on a physical iOS device, consider creating a [development build](/develop/development-builds/introduction.mdx).

Reviewer: ```suggestion
If you can't upgrade your project to a newer SDK version, you can still run it using an older version of Expo Go. However, this method will not work for physical iOS devices because it is not possible to install apps directly (sideload). If you need to run your project on a physical iOS device, consider creating a [development build](/develop/development-builds/introduction).
```

- Let's break the sentence here.

> If you are trying to use this method on a physical Android device, you will need to connect it to your computer and enable the **USB Debugging** option. Learn more about [Android emulators](/workflow/android-studio-emulator.mdx) and [iOS simulators](/workflow/ios-simulator.mdx)

Reviewer: ```suggestion
If you are trying to use this method on a physical Android device, connect it to your computer and enable the **USB Debugging** option. To learn more, see [Android emulators](/workflow/android-studio-emulator) and [iOS simulators](/workflow/ios-simulator)
```

We release a new SDK version approximately every quarter. Find out which [versions of React Native map to which versions of the Expo SDK](/versions/latest/#each-expo-sdk-version-depends-on-a-react-native-version).

#### Upgrading Your Project to a New SDK Version

> **Info** We recommend upgrading SDK versions incrementally, one at a time. Doing so will help you pinpoint breakages and issues that arise during the upgrade process.

To upgrade your project, you should:
- Update to the latest version of the EAS CLI (if you use it):\
`npm i -g eas-cli`
- Install the new version of the Expo package:\
`npm install expo@^<sdk_version>` or `yarn add expo@^<sdk_version>` (e.g. `yarn add expo@^49.0.0`)
- Upgrade all dependencies to match the new SDK:\
`npx expo install --fix`
- Check for any possible issues in your project dependencies:\

> `npx expo-doctor@latest`

Author: We are recommend updating incrementally, so using `expo@lastest` won't be helpful if someone needs to update by more than one major version

> `npx expo-doctor@latest`

Reviewer: That makes sense. My bad, I completely forgot about the aspect of incremental upgrading.

> If you encounter any issues while upgrading, refer to the [blog post](/workflow/upgrading-expo-sdk-walkthrough/) about your new SDK. Release blog posts contain more details about the update process for each SDK version. Also, blog posts include information on any breaking changes introduced in particular release.

Reviewer: ```suggestion
If you encounter any issues while upgrading, see [blog post](/workflow/upgrading-expo-sdk-walkthrough/) about the SDK version. Release blog posts contain more details about the update process for each SDK version. Also, blog posts include information on any breaking changes introduced in a particular release.
```

One grammatical mistake and updated verbiage for more clarity

#### Running an Unsupported Project Using an Older Version of Expo Go

If you can't upgrade your project to a newer SDK version, you can still run it using an older version of Expo Go:
- If installed, remove Expo Go from your device/simulator.
- Start the development server using `yarn start` or `npx expo start` and wait for it to load.
- Press `i` or `a` to run your app on `iOS` or `Android` respectively.
- Compatible Expo Go will be automatically installed on your device/simulator and your project will open.

> > Note: This method will not work for physical iOS devices. If you are trying to use this method on a physical Android device, you should plug it in with a USB cable and have `USB Debugging` turned on. If you want to learn more about the Android Emulator and iOS Simulator, read the [Login into Expo Go section](#login-into-expo-go).

Reviewer: ```suggestion
> **Note:** This method will not work for physical iOS devices. If you are trying to use this method on a physical Android device, plug it in with a USB cable and have the **USB Debugging** option turned on. To learn more about the Android Emulator and iOS Simulator, see [Login into Expo Go section](#login-into-expo-go).
```

  <Tabs>
    <Tab label="npm">
      <Terminal cmd={['$ npm install expo@^<sdk_version>']} />
    </Tab>
    <Tab label="yarn">
      <Terminal cmd={['$ yarn add expo@^<sdk_version>']} />
    </Tab>
  </Tabs>
</Step>
<Step label="3">
  Upgrade all dependencies to match the new SDK:
  <Terminal cmd={['$ npx expo install --fix']} />
</Step>
<Step label="4">
  Check for any possible issues in your project dependencies:
  <Terminal cmd={['$ npx expo-doctor@latest']} />
</Step>

If you encounter any issues while upgrading, see [blog post](/workflow/upgrading-expo-sdk-walkthrough/) about the SDK version. Release blog posts contain more details about the update process for each SDK version. Also, blog posts include information on any breaking changes introduced in a particular release.

#### Run an unsupported project in an older version of Expo Go

If you can't upgrade your project to a newer SDK version, you can still run it using an older version of Expo Go:
<Step label="1">
  If installed, remove Expo Go from your device/simulator.
</Step>
<Step label="2">
  Start the development server:
  <Terminal cmd={['npx expo start']}/>
</Step>
<Step label="3">

>   Press <kbd>a</kbd> or <kbd>i</kbd> to run your app on Android or iOS.

Reviewer: ```suggestion
  Press <kbd>a</kbd> or <kbd>i</kbd> to run your app on Android or iOS. This will download and install the compatible Expo Go on your device/simulator and open your project.
```

</Step>

> Compatible Expo Go will be automatically installed on your device/simulator, and your project will open.

Reviewer: ```suggestion
```

Let's remove this since this step looks like it will only happen when the Step 3 becomes an action. So let's move this with step 3.
